org.adempiere.base/src/org/adempiere/model/GenericPO.java

```diff
@@ -132,6 +146,604 @@ protected int get_AccessLevel() {
+	 * @return
+	 */
+	private boolean columnExists(String columnName, boolean throwException) {
+		int idx = get_ColumnIndex(columnName);
+		if (idx < 0 && throwException)
+			throw new AdempiereException("Column " + get_TableName() +"." + columnName + " not found");
+		return (idx >= 0);
+	}
+
+	/**
+	 * Verify if a column exists
+	 * @param columnName
+	 * @return boolean
+	 */
+	private boolean columnExists(String columnName) {
+		return columnExists(columnName, false);
+	}
+
+	/** Column name DocAction */
+	public static final String COLUMNNAME_DocAction = "DocAction";
+	/** Column name DocStatus */
+	public static final String COLUMNNAME_DocStatus = "DocStatus";
+	/** Column name Processing */
+	public static final String COLUMNNAME_Processing = "Processing";
+	/** Column name C_DocTypeTarget_ID */
+	public static final String COLUMNNAME_C_DocTypeTarget_ID = "C_DocTypeTarget_ID";
+	/** Column name C_DocType_ID */
+	public static final String COLUMNNAME_C_DocType_ID = "C_DocType_ID";
+	/** Column name DateAcct */
+	public static final String COLUMNNAME_DateAcct = "DateAcct";
+	/** Column name DateTrx */
+	public static final String COLUMNNAME_DateTrx = "DateTrx";
+	/** Column name IsApproved */
+	public static final String COLUMNNAME_IsApproved = "IsApproved";
+	/** Column name Processed */
+	public static final String COLUMNNAME_Processed = "Processed";
+	/** Column name DocumentNo */
+	public static final String COLUMNNAME_DocumentNo = "DocumentNo";
+	/** Column name Description */
+	public static final String COLUMNNAME_Description = "Description";
+	/** Column name SalesRep_ID */
+	public static final String COLUMNNAME_SalesRep_ID = "SalesRep_ID";
+	/** Column name AD_User_ID */
+	public static final String COLUMNNAME_AD_User_ID = "AD_User_ID";
+	/** Column name C_Currency_ID */
+	public static final String COLUMNNAME_C_Currency_ID = "C_Currency_ID";
+
+	/** DocStatus AD_Reference_ID=131 */
+	public static final int DOCSTATUS_AD_Reference_ID=131;
+	/** Drafted = DR */
+	public static final String DOCSTATUS_Drafted = "DR";
+	/** Completed = CO */
+	public static final String DOCSTATUS_Completed = "CO";
+	/** Approved = AP */
+	public static final String DOCSTATUS_Approved = "AP";
+	/** Not Approved = NA */
+	public static final String DOCSTATUS_NotApproved = "NA";
+	/** Voided = VO */
+	public static final String DOCSTATUS_Voided = "VO";
+	/** Invalid = IN */
+	public static final String DOCSTATUS_Invalid = "IN";
+	/** Reversed = RE */
+	public static final String DOCSTATUS_Reversed = "RE";
+	/** Closed = CL */
+	public static final String DOCSTATUS_Closed = "CL";
+	/** Unknown = ?? */
+	public static final String DOCSTATUS_Unknown = "??";
+	/** In Progress = IP */
+	public static final String DOCSTATUS_InProgress = "IP";
+	/** Waiting Payment = WP */
+	public static final String DOCSTATUS_WaitingPayment = "WP";
+	/** Waiting Confirmation = WC */
+	public static final String DOCSTATUS_WaitingConfirmation = "WC";
+	/**

```

Comment:
why the need to duplicate the constants at org.compiere.process.DocAction ?

Reply:
done